snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -811,10 +814,46 @@ def _get_primed_stage_packages(self, snap_files: Set[str]) -> Set[str]:
                 primed_stage_packages.add(stage_package)
         return primed_stage_packages
 
+    def split_debug_info(self, file_paths: Set[str]) -> Tuple[Set[str], Set[str]]:
+        logger.warn("*EXPERIMENTAL*: Splitting debug information!")
+
+        debug_files: Set[str] = set()
+        debug_dirs: Set[str] = set()
+
+        debug_dir = Path(self.primedir, "usr", "lib", "debug", ".build-id")
```

Comment:
I would make this Path(project.workdir) + "debug" or tempfile.mkdtemp of tempfile.TemporaryDirectory. Also mention where it is being put in the logger.warn (which should be logger.warning, https://docs.python.org/3/library/logging.html#logging.Logger.warning)